### PR TITLE
Make Action.enabled be isEnabled(Intent intent) instead.

### LINKED ISSUE #53276: Update Actions API to not need FocusNodes and improve Intent->Action mapping
We've had some feedback that people would like to be able to invoke actions when they are not in the context of keyboard input (from a mouse click, etc.).

This was part of the original design (e.g. the focus node was always optional in ActionDispatcher.invokeAction), but the invoke signature only has a focus node, so one had to be supplied. Since the focus node that was provided was nearly always resolving to `primaryFocus`, we should eliminate the `focusNode` parameter from the `invoke`, as it isn't needed, and add a new `Action` abstract subclass called `ContextAction` which can supply a `BuildContext` to `invoke` if available.

It's also the case that the IntentKey argument to the Action base class is never used, and so should be eliminated.

There is a design doc for the changes needed: flutter.dev/go/actions-and-shortcuts-design-revision

This is a tracking PR for that design.

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

In order to allow actions to take into account the Intent type they are associated with when they evaluate whether they are enabled or not, modify the `enabled` accessor to take an Intent.

This means that the `Actions` widget will trigger a rebuild for those widgets which have an `InheritedWidget` dependency upon it whenever any action in its map changes, since it can't ask specifically about the `enabled` status without an `Intent` instance. In practice, however, this isn't actually all that different from the way it works now, since most actions will only notify for enabled changes, and any enabled change triggered a rebuild anyhow.  This just broadens the rebuilds to include all state changes that might occur in an `Action`, which is desirable anyhow.

## Related Issues

- Fixes https://github.com/flutter/flutter/issues/53276

## Tests

- Modified tests to use `isEnabled` instead of `enabled`

## Breaking Change

- [X] No, this is *not* a breaking change.

### CODE REVIEW

**Review by krisgiesing — APPROVED** [review]
This change LGTM. You may also want goderbauer's review before landing, but from an API consumption perspective, the looks good.

File: packages/flutter/lib/src/widgets/actions.dart
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/actions.dart:652]: Wrap this in setState to ensure it rebuilds?

**Review by goderbauer — APPROVED** [review]
LGTM after comment is addressed.

File: packages/flutter/lib/src/widgets/actions.dart
Comment (gspencergoog) [reply packages/flutter/lib/src/widgets/actions.dart:652]: Thanks, done.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**
